IRCT-CL/src/main/java/edu/harvard/hms/dbmi/bd2k/irct/cl/util/Utilities.java

```diff
@@ -259,4 +273,44 @@ public static String extractUserFromTokenIntrospection(HttpServletRequest req, S
 
 		return null;
 	}
+
+	/**
+	 * to apply a proxy to a http function by reading the environment variables from JVM
+	 * @param request
+	 */
+	public static void applyProxySettings(HttpRequestBase request) {
+		String proxyHost = System.getProperty("http.proxyHost");
+		String proxyPort = System.getProperty("http.proxyPort");
+		String proxyProtocol = System.getProperty("http.proxyProtocol"); // non-standard
+		if (proxyHost != null) {
+			int port = 80;
+			if (proxyPort != null) {
+				port = Integer.valueOf(proxyPort);
+			}
+
+			if (proxyProtocol == null) {
+				proxyProtocol = "http";
+				if (port == 443) {
+					proxyProtocol = "https";
+				}
+			}
```

Comment:
What if `proxyProtocol` is not null and it has value as `http` and `port` has value `443`. So, don't you want to make  `proxyProtocol` change to `https` ?

Reply:
To my view, this will leave to users to make sure it matches. Since proxy schema can match to any ports based on how the server configs, only default port for https is 443, and for http is 80.